src/commands/transaction/mosaicaddressrestriction.ts

```diff
@@ -0,0 +1,147 @@
+        description: '(Optional) Previous restriction value.',
+    })
+    previousRestrictionValue: string;
+
+    @option({
+        flag: 'V',
+        description: 'New restriction value',
+    })
+    newRestrictionValue: string;
+
+    @option({
+        flag: 'T',
+        description: 'New restriction type',
+        validator: new MosaicRestrictionType(),
+    })
+    newRestrictionType: number;
+}
+
+@command({
+    description: 'Set a mosaic restriction to an specific address.',
+})
+export default class extends ProfileCommand {
+    constructor() {
+        super();
+    }
+    @metadata
+    execute(options: CommandOptions) {
+        options.mosaicid = OptionsResolver(
+            options,
+            'mosaicid',
+            () => undefined,
+            'The identifier of the mosaic being restricted: ',
+        );
+
+        options.targetAddress = OptionsResolver(
+            options,
+            'targetAddress',
+            () => undefined,
+            'The address being restricted: ',
+        );
+
+        options.restrictionKey = OptionsResolver(
+            options,
+            'restrictionkey',
+            () => undefined,
+            'Restriction key relative to the reference mosaic identifier: ',
+        );
+
+        options.newRestrictionValue = OptionsResolver(
+            options,
+            'newRestrictionValue',
+            () => undefined,
+            'Restriction newRestrictionValue: ',
+        );
+
+        options.newRestrictionType = parseInt(OptionsResolver(
+            options,
+            'newRestrictionType',
+            () => undefined,
+            'Restriction newRestrictionType: ',
+        ), 10);
+
+        if (!CommandOptions.limitType.includes(options.newRestrictionType)) {
+            throw new ExpectedError('Wrong mosaic restriction type');
+        }
+
+        console.log(options);
+        const profile = this.getProfile(options);
+
+        const addressRestrictionTransaction = MosaicAddressRestrictionTransaction.create(
+            Deadline.create(),
+            new MosaicId(options.mosaicid),
+            UInt64.fromNumericString(options.restrictionKey),
+            Address.createFromRawAddress(options.targetAddress),
+            UInt64.fromNumericString(options.newRestrictionValue),
+            profile.networkType,
+            options.previousRestrictionValue ? UInt64.fromNumericString(options.previousRestrictionValue) : undefined,
+            UInt64.fromUint(options.maxFee),
+        );
+
+        const account = Account.createFromPrivateKey(profile.account.privateKey, profile.networkType);
+        const networkGenerationHash = profile.networkGenerationHash;
+
+        const signedTransaction = account.sign(addressRestrictionTransaction, networkGenerationHash);
+        console.log(chalk.green('signed transaction hash: \n'));
+        console.log(signedTransaction.hash + '\n');
+
+        const transactionHttp = new TransactionHttp(profile.url);
```

Comment:
Replace using `` this.announceTransaction(signedTransaction, profile.url)`` function. Example: https://github.com/nemtech/nem2-cli/blob/master/src/commands/transaction/mosaic.ts#L167